### ISSUE TITLE
Sticky Table Horizontal & Vertical Row & Column on Question/Dashboard

### ISSUE THREAD

**Issue by texamania** [metabase/metabase#13799]
I want use the sticky headers.

**Comment by PauloHFM** [comment]
Hi @flamber, no milestone added this yet,right? it's on the roadmap? 
I thought  #14177 would include this!

Thanks very much!

**Comment by flamber** [comment]
@PauloHFM This issue is for manually freezing - and it [was requested](https://discourse.metabase.com/t/immobilize-columns/12280?u=flamber) for regular tables from what I can understand, but who knows, when there isn't more details.
The new Pivot Table coming in 0.38 has frozen headers and columns.
![image](https://user-images.githubusercontent.com/1447303/107397772-3d256e00-6aff-11eb-9c72-29a357484e41.png)

**Comment by PauloHFM** [comment]
That's nice! the "frozen" headers up and left side on the image I think it's what everybody needs! 
Thanks!

**Comment by PauloHFM** [comment]
@flamber I think the other option would be freeze the totals on the right and down sides of the pivot table. Could be a toogle on pivot tables options!

**Comment by flamber** [comment]
@PauloHFM Please open a separate request for that - you can reference this issue. But from what I understand this issue is a request for plain tables.

**Comment by mlm603** [comment]
The ability to make the first column sticky would be really useful for us on regular table vizzes, especially since the Pivot Table viz is not an option for Native Questions.

**Comment by basquens-talentify** [comment]
Hello, any updates about this?

**Comment by mattheworiordan** [comment]
Fixed columns would definitely be a big win. We have very wide tables making metabase unusable for these cases, and exporting to Excel/Google Sheets is not ideal. Hoping a fix can be released for this.

**Comment by yash-dentolo** [comment]
Love the new metabase updates! Wishing they had this one as well :/

**Comment by oliviertassinari** [comment]
I would love to see this feature land. I have been frustrated a couple of times about this when using Metabase, it tends to be a popular request, e.g. https://github.com/mui/mui-x/issues/193.

**Comment by hbd** [comment]
+1 on a "freeze up to X column" request!

**Comment by levon-sib** [comment]
Hi, any news?

**Comment by SeshTiliRest** [comment]
Hi! I think many people are waiting for this feature to appear 🙏

**Comment by ben-nob** [comment]
Is there any update on this?

**Comment by malakmet** [comment]
Would love this feature! I think it would really help if it was the same like on Excel and Sheets!

**Comment by AhmedSameh90** [comment]
Really needed!

**Comment by thunguyenthu** [comment]
We really need it!

**Comment by cdeweyx** [comment]
Basic version could be a column formatting setting so you could freeze columns as you please, maybe if you freeze it would auto-move the column to the far left with any other sticky columns, then you could reorder within that constraint.

**Comment by siddharthbulia** [comment]
Yes, this is really needed!

**Comment by BadrBensassi** [comment]
@flamber Did this feature reach the roadmap yet? Thanks for the help

**Comment by cdeweyx** [comment]
> [@flamber](https://github.com/flamber) Did this feature reach the roadmap yet? Thanks for the help

Yes this feature is planned for the next release or two. It's bundled in the "Better table visualizations" item on the roadmap.<img width="710" alt="Image" src="https://github.com/user-attachments/assets/861368ac-9cee-4089-95e4-e3e99f5d9ca9" />

**Comment by shahrzadpeyman** [comment]
Yes, this is really needed!

**Comment by bartdegroot92** [comment]
This feature would be great!

**Comment by mundoc** [comment]
Hi! Any news on this? Do we know if there is an ETA?

**Comment by banditelol** [comment]
> > [@flamber](https://github.com/flamber) Did this feature reach the roadmap yet? Thanks for the help
> 
> Yes this feature is planned for the next release or two. It's bundled in the "Better table visualizations" item on the roadmap.<img alt="Image" width="710" src="https://private-user-images.githubusercontent.com/16455495/396153207-861368ac-9cee-4089-95e4-e3e99f5d9ca9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkwMDI4MTMsIm5iZiI6MTc0OTAwMjUxMywicGF0aCI6Ii8xNjQ1NTQ5NS8zOTYxNTMyMDctODYxMzY4YWMtOWNlZS00MDg5LTk1ZTQtZTNlOTlmNWQ5Y2E5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjA0VDAyMDE1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExNjMxMzBjMTk3YzNiNGI5NjU1NmIwODM5OTUxNTYyYjEyMzU2NmY0OGQ3YTM2OWZiMmNhMzVkZTY3MWRlMTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FjgOWhpHGT0lN0XA4t9oFyY-JiSfrbWVhsJfFQFNLrk">

Hello, I see that this [has been released](https://www.metabase.com/releases/metabase-54#better-table-visualization), but there's no column freezing yet is there a change of plan or would it be planned for the minor update? 

thank you

**Comment by poorbadger** [comment]
This is an important feature for us - our users tend towards tabular format and going beyond 6 or so columns results in the loss of context as you scroll across.

**Comment by thuthuynt** [comment]
Hello, I believe this feature is definitely needed, our users from multiple companies have been asking for it repeatedly. However, I no longer see it on the Metabase product roadmap(https://www.metabase.com/roadmap). 
Please let us know if you still plan to develop this feature in the near future. Thank you!

**Comment by canay-f** [comment]
this feature is essential and easy to implement

**Comment by duf59** [comment]
This would be a very useful feature, currently I'm sticking to google sheet/excel for some tables because of this.

**Comment by dominykaszenk** [comment]
Just want to echo what others have said before - having this feature would make wide tables a lot easier to use

**Comment by alexandreharoutunian-acseo** [comment]
Given the amount of interest this has gathered over the years, is it reasonable to hope this might land sometime before 2026, or should we keep our expectations… flexible?

**Comment by banditelol** [comment]
Hello @cdeweyx @dragonsahead , I've been tinkering around with the idea for a while in [my repo ](https://github.com/banditelol/metabase-pin-column)(I'm planning on using it internally on my own)

I was thinking it could be implemented without additional dependencies, but there are some cases that I couldn't decide what's the correct behaviour, for example:
1. Where to put the option? 
    - I've tried putting it inside the popup but it's not clear on a glance which columns are pinned. So I tried adding an icon beside the column name which so far work the best.
2. Should we allow right pinning (I've played around with this idea, but cant settle on good behavior)?
3. How many columns should be able to be frozen (currently I'm thinking of at most 3)?
4. Should I allow Long text to be pinned? if yes should I limit the column width automatically?

and some more. If You are open for contribution and discussion I'm interested in working on it :)

* Disclaimer I create it with help of Amp CLI. And has read & test the feature myself

https://github.com/user-attachments/assets/c171eef9-3a52-4112-915c-04a2030ef92e

**Comment by louisrae** [comment]
Echoing support for this - massive value add

**Comment by mattheworiordan** [comment]
> If You are open for contribution and discussion I'm interested in working on it :)

I've been following this issue for a few years now. I'd love to see this, but I am afraid based on my experience of contributing to Metabase, you're unlikely to land this. [I went back and forth on a small feature improvement for 5 months and 7 iterations in response to their asks and change in direction](https://github.com/metabase/metabase/pull/62235#issuecomment-3767032238) only to have the PR closed and an issue added to the 3.6k backlog!

Good luck, I hope this lands, but don't hold your breath
